### PR TITLE
Password doors.

### PR DESCRIPTION
**Pull request by AnturK**
Simple password doors for admin abuse/ruins.

### CODE REVIEW

**Comment by WJohn** [comment]
Awesome!

**Comment by AnturK** [comment]
Right now it's it's map/varedit but adding randomized subtype is easy enough.

**Comment by AlexTheSysop42** [comment]
can they be built by crewmembers?

**Comment by JJRcop** [comment]
@AlexTheSysop42 This pr just adds the door, no build recipes

**Comment by coiax** [comment]
The door should also accept input verbally. Unless the door has a keyboard sprite?

File: code/game/machinery/doors/passworddoor.dm [hunk 1]
```diff
@@ -0,0 +1,58 @@
+/obj/machinery/door/password
+	name = "door"
+	desc = "This door only opens when provided a password."
+	icon = 'icons/obj/doors/blastdoor.dmi'
+	icon_state = "closed"
+	explosion_block = 3
+	heat_proof = TRUE
+	max_integrity = 600
+	armor = list(melee = 100, bullet = 100, laser = 100, energy = 100, bomb = 100, bio = 100, rad = 100, fire = 100, acid = 100)
```
Comment (Cyberboss) [inline code/game/machinery/doors/passworddoor.dm]: quotes

File: code/game/machinery/doors/passworddoor.dm [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+	resistance_flags = INDESTRUCTIBLE | FIRE_PROOF
+	damage_deflection = 70
+	var/password = "Swordfish"
+
+/obj/machinery/door/password/CollidedWith(atom/movable/AM)
+	if(density)
+		return 0
+	else
+		return ..()
```
Comment (Cyberboss) [inline code/game/machinery/doors/passworddoor.dm]: return !density && ..()

**Review by Cyberboss — CHANGES_REQUESTED** [review]

**Comment by AnturK** [comment]
@coiax Added voice activated version/var. This uses the blast door sprite right now. Mostly because i imagine styled subtypes for specific ruins will be needed anyway.

File: code/game/machinery/doors/passworddoor.dm [hunk 3]
```diff
@@ -0,0 +1,72 @@
+/obj/machinery/door/password
+	name = "door"
+	desc = "This door only opens when provided a password."
+	icon = 'icons/obj/doors/blastdoor.dmi'
+	icon_state = "closed"
+	explosion_block = 3
+	heat_proof = TRUE
+	max_integrity = 600
+	armor = list(melee = 100, bullet = 100, laser = 100, energy = 100, bomb = 100, bio = 100, rad = 100, fire = 100, acid = 100)
+	resistance_flags = INDESTRUCTIBLE | FIRE_PROOF
```
Comment (ShizCalev) [inline code/game/machinery/doors/passworddoor.dm]: ACID_PROOF?

**Pull request merged**